controllers/common/fx.go

```diff
@@ -145,3 +153,54 @@ func Bootstrap(params Params) error {
 
 	return nil
 }
+
+// PickChildCRDPredicate allows events to trigger the Reconcile of Chaos CRD,
+// for example:
+// Reconcile of IOChaos could be triggered by changes on PodIOChaos.
+// For now, we have PodHttpChaos/PodIOChaos/PodNetworkChaos which require to follow this pattern.
+type PickChildCRDPredicate struct {
+	predicate.Funcs
+}
+
+// Update implements UpdateEvent filter for child CRD.
+func (PickChildCRDPredicate) Update(e event.UpdateEvent) bool {
+	switch e.ObjectNew.(type) {
+	case *v1alpha1.PodHttpChaos, *v1alpha1.PodIOChaos, *v1alpha1.PodNetworkChaos:
+		return true
+	}
+	return false
+}
+
+// StatusRecordEventsChangePredicate skip the update event,
+// when we Only update object.status.experiment.records[].events
+type StatusRecordEventsChangePredicate struct {
+	predicate.Funcs
+}
+
+// Update implements UpdateEvent filter for update deletionTimestamp.
```

Comment:
It seems that the comment is not related with the implementation? It has nothing to do with `deletionTimestamp`

Reply:
oposs ... I forget to update this... `deletionTimestamp` will introduced by this comment https://github.com/chaos-mesh/chaos-mesh/pull/3170#issuecomment-1120231330